### PR TITLE
Add version to Info Menu

### PR DESCRIPTION
**Pull request by bbsan2k**
This PR adds some version information to a new Info menu.

The version is generated based on the newest git tag that is not "latest".

Additionally it outputs some information about whether USB Debug flag was set or not and about the commit id, the firmware has been built with.

Note: To have a meaningful versioning it is required to add git tags if a new release is done.

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -21,6 +21,15 @@ jobs:
     - uses: actions/checkout@v3
       with:
          submodules: 'recursive'
+         fetch-depth: '0'
+    - run: |
+        git fetch --prune --tags
+        git tag --list
+        git fetch --all --tags
+        SD2PSX_VERSION=$(git describe --exclude latest | sed 's/\(.*\)-.*/\1/')
```
Comment (xyzz) [inline .github/workflows/cmake.yml:29]: I think this should go in a separate pr. seems it would require annotated tags? and what happens for automatic builds for commits which aren't tagged? tbh once we start tagging releases I'd rather post them myself, rather than it being automatic

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@ build
 .vscode
 keys.c
 .cache
-gamedbps*.dat
\ No newline at end of file
+gamedbps*.dat
+src/version/version.c
```
Comment (xyzz) [inline .gitignore:6]: as below

File: src/version/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+find_package(Git QUIET)
+
+if(Git_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/../../.git")
+    execute_process(COMMAND ${GIT_EXECUTABLE} fetch --all --tags --force)
+    execute_process(COMMAND ${GIT_EXECUTABLE} describe --exclude latest --abbrev=0
```
Comment (xyzz) [inline src/version/CMakeLists.txt:6]: again not sure it's good idea to use annotated tags over lightweight + what happens for builds several commits after tagged version

File: src/version/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    string(REGEX REPLACE "\n$" "" SD2PSX_COMMIT "${SD2PSX_COMMIT}")
+    string(REGEX REPLACE "\n$" "" SD2PSX_BRANCH "${SD2PSX_BRANCH}")
+else()
+    set(SD2PSX_VERSION "None")
+    set(SD2PSX_COMMIT "None")
+    set(SD2PSX_BRANCH "None")
+endif()
+    
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/template/version.c ${CMAKE_CURRENT_SOURCE_DIR}/version.c @ONLY)
```
Comment (xyzz) [inline src/version/CMakeLists.txt:22]: lets not create outputs under the source dir - put it under build dir instead. as an alternative, can keep the C file and have it #include an auto-generated .h file (in case it's difficult to compile things under build directly)

**Review by xyzz — CHANGES_REQUESTED** [review]

File: src/version/CMakeLists.txt (hunk 4 above)
Comment (bbsan2k) [reply src/version/CMakeLists.txt]: Moved to build dir

File: src/version/CMakeLists.txt (hunk 3 above)
Comment (bbsan2k) [reply src/version/CMakeLists.txt:6]: See latest release in my fork. Increasing number is added after dash.

File: .github/workflows/cmake.yml [hunk 5]
```diff
@@ -21,6 +21,23 @@ jobs:
     - uses: actions/checkout@v3
       with:
          submodules: 'recursive'
+         fetch-depth: '0'
+    - run: |
+        git fetch --prune --tags
+        git tag --list
+        git fetch --all --tags
```
Comment (xyzz) [inline .github/workflows/cmake.yml:28]: what's the purpose of double git-fetch? what's the purpose of --prune, since it's a fresh checkout I don't think it's needed? Also, wouldn't it get all of the recent tags anyway when it's cloning? fetch iirc only needed if you have an older repo and just doing git-pulls

File: src/version/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,25 @@
+
+find_package(Git QUIET)
+
+if(Git_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/../../.git")
+    execute_process(COMMAND ${GIT_EXECUTABLE} fetch --all --tags --force)
```
[the remark this answers is not in the document]
Comment (bbsan2k) [reply src/version/CMakeLists.txt]: The --force should overwrite local tags if there are already remote tags with the same name. It should not overwrite local changes AFAIK

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (bbsan2k) [reply .github/workflows/cmake.yml]: Fixed - now it either uses the Tag as the version or created a build called ```nightly-<<COMMIT_HASH>>```

File: src/gui.c [hunk 7]
```diff
@@ -589,6 +591,26 @@ static void create_menu_screen(void) {
         lv_obj_add_event_cb(cont, evt_do_civ_deploy, LV_EVENT_CLICKED, NULL);
     }
 
+    /* Info submenu */
+    lv_obj_t *info_page = ui_menu_subpage_create(menu, "Info");
+    {
+        cont = ui_menu_cont_create_nav(info_page);
+        ui_label_create_grow_scroll(cont, "Version");
```
[the remark this answers is not in the document]
Comment (bbsan2k) [reply src/gui.c:598]: Fixed - Version now only says "nightly" - commit ID below

File: .github/workflows/cmake.yml (hunk 5 above)
Comment (bbsan2k) [reply .github/workflows/cmake.yml]: Is now simplified

File: src/version/CMakeLists.txt (hunk 6 above)
[the remark this answers is not in the document]
Comment (xyzz) [reply src/version/CMakeLists.txt]: Okay even if force doesn't touch anything bug tags, this doesn't belong during configure or build step. lets follow principle of least astonishment, a random git fetch against current repo during build or configure is not it.
